OJS.Workers.ExecutionStrategies/BaseCodeExecutionStrategy.cs

```diff
@@ -40,13 +40,17 @@ protected virtual string SaveCodeToTempFile<TINput>(IExecutionContext<TINput> ex
                 ? FileHelpers.SaveStringToTempFile(this.WorkingDirectory, executionContext.Code)
                 : FileHelpers.SaveByteArrayToTempFile(this.WorkingDirectory, executionContext.FileContent);
 
-        protected void SaveZipSubmission(byte[] submissionContent, string directory)
+        protected void SaveZipSubmission(byte[] submissionContent, string directory, bool deleteZip = true)
         {
             var submissionFilePath = FileHelpers.BuildPath(directory, ZippedSubmissionName);
             FileHelpers.WriteAllBytes(submissionFilePath, submissionContent);
             FileHelpers.RemoveFilesFromZip(submissionFilePath, RemoveMacFolderPattern);
             FileHelpers.UnzipFile(submissionFilePath, directory);
-            FileHelpers.DeleteFile(submissionFilePath);
+
+            if (deleteZip)
+            {
+                FileHelpers.DeleteFile(submissionFilePath);
+            }
```

Comment:
Why?

Reply:
Removed unzipping of file entirely as its not needed. The compilator compiles the zip and  deletes the source file automatically.